lib/index.js

```diff
@@ -86,35 +86,32 @@ module.exports = function(format, options, formatterOptions) {
       cb();
     },
     function(cb) {
-      documentation.build(
-        files.map(function(file) {
-          return file.path;
-        }),
-        options,
-        function(err, comments) {
-          formatter(
-            comments,
-            formatterOptions,
-            function(err, output) {
-              if (format === 'json' || format === 'md') {
-                this.push(
-                  new File({
-                    path: options.filename || 'API.' + format,
-                    contents: new Buffer(output)
-                  })
-                );
-              } else if (format === 'html') {
-                output.forEach(
-                  function(file) {
-                    this.push(file);
-                  }.bind(this)
-                );
-              }
-              cb();
-            }.bind(this)
-          );
-        }.bind(this)
-      );
+      documentation
+        .build(
+          files.map(function(file) {
+            return file.path;
+          }),
+          options
+        )
+        .then(comments => {
+          formatter(comments, formatterOptions).then(output => {
+            if (format === 'json' || format === 'md') {
+              this.push(
+                new File({
+                  path: options.filename || 'API.' + format,
+                  contents: new Buffer(output)
+                })
+              );
+            } else if (format === 'html') {
+              output.forEach(
+                function(file) {
```

Comment:
This PR starts using arrow functions, which seems like a good idea since most folks are on Node >= 4 anyway. So in that case, should be possible to use an arrow function here and avoid having to use `.bind(this)`